packages/x-charts/src/ChartsReferenceLine/ChartsXReferenceLine.tsx

```diff
@@ -0,0 +1,96 @@
+import * as React from 'react';
+import { useDrawingArea, useXScale } from '../hooks';
+import {
+  CommonChartsReferenceLineProps,
+  ReferenceLineRoot,
+  getReferenceLineClasses,
+} from './common';
+import { ChartsText } from '../internals/components/ChartsText';
+
+export type ChartsXReferenceLineProps<
+  TValue extends string | number | Date = string | number | Date,
+> = CommonChartsReferenceLineProps & {
+  /**
+   * The x value associated to the reference line.
+   * If defined the reference line will be vertical.
+   */
+  x: TValue;
+};
+
+type GetTextPlacementParams = {
+  top: number;
+  height: number;
+} & Pick<CommonChartsReferenceLineProps, 'labelAlign'>;
+
+const getTextParams = ({ top, height, labelAlign = 'middle' }: GetTextPlacementParams) => {
+  switch (labelAlign) {
+    case 'start':
+      return {
+        y: top,
+        style: {
+          dominantBaseline: 'hanging',
+          textAnchor: 'start',
+        } as const,
+      };
+
+    case 'end':
+      return {
+        y: top + height,
```

Comment:
IMHO, we should solve this problem:
![Screenshot 2023-11-01 at 13 01 00](https://github.com/mui/mui-x/assets/4941090/e7d5288b-5a15-4bc2-918d-ab0d935971e0)
At least changing the `dominantBaseline` to `ideographic` avoids the label overlapping the `ChartsXAxis`.
Or maybe it would also make sense to introduce or change the `spacing` prop to accept the `x` and `y` prop so that the end user could tailor the result to their needs?